### PR TITLE
Fixes for macros for llvm and extern blocks

### PR DESCRIPTION
**Pull request by mppf**
* Enables --llvm to work with GMP
* fixes test cases just added to do with #defines refering to other #defines

Fixed the case when a macro is a C declaration already existing in the layered
value table - now we add an entry to the table from the macro name to the C
declaration. In order to do that, needed to add a new addGlobalCDecl that takes
in a name to use (instead of the name in a C NamedDecl).  Also, since now
handleMacro might add one of two different types to the layered value table,
it's harder for it to return the added value. Instead of returning a value, now
handleMacro returns void. A caller wishing to find the value it added should
just query the layer value table.

(The layered value table is a cname ->  Clang/LLVM thing map used within
 LLVM code generation and with extern block support).

test/extern/ferguson/externblock and test/modules/standard/gmp/ferguson
both pass with and without --llvm; full testing passes with C backend;
and LLVM testing works for release/ and extern/ directories.

### CODE REVIEW

**Comment by mppf** [comment]
passed full local C backend testing

**Comment by mppf** [comment]
Testing release and extern with --llvm

**Comment by mppf** [comment]
those tests passed

File: compiler/util/clangUtil.cpp [hunk 1]
```diff
@@ -946,6 +992,9 @@ void runClang(const char* just_parse_filename) {
     // Now initialize a code generator...
     // this will enable us to ask for addresses of static (inline) functions
     // and cause them to be emitted eventually.
+    // CCodeGenAction is defined above. It traverses the C AST
+    // and does the code generation.
+    //printf("running CCodeGenAction\n");
```
Comment (ronawho) [inline compiler/util/clangUtil.cpp]: Should this either be removed or wrapped in an `if( debugPrint) {}`?

**Comment by ronawho** [comment]
What happens if a macro that is too complex is encountered?

File: compiler/util/clangUtil.cpp (hunk 1 above)
Comment (mppf) [inline compiler/util/clangUtil.cpp]: debugPrint doesn't exist here. I'd be happy to remove it though.

**Comment by mppf** [comment]
A too-complex macro will not be added to the Layered Value Table. When that macro is used (during code generation or in an extern block) there will be no entry for it and the error message indicates that fancy macros could be the source of the problem.

**Comment by ronawho** [comment]
Is it worth adding a test for that and/or adding error messages that explicitly say the macro is too complex? (I have no idea, so I'll defer to whatever you think is best.)

**Comment by mppf** [comment]
The issue is that a too-complex macro might not be used, since the normal use case is extern { #include "giant-header-file.h" }. So we don't want to generate an error in handleMacro. We *could* add an entry to the LayeredValueTable to indicate that the macro was too complex and it should be an error... but that would take a little bit of work and I don't know that the current situation is so bad.

**Comment by ronawho** [comment]
Ah ok, got it. I'm fine with the current situation then, thanks for the explanation.

File: compiler/util/clangUtil.cpp (hunk 1 above)
Comment (ronawho) [inline compiler/util/clangUtil.cpp]: Yeah, I think that'd be good

**Comment by ronawho** [comment]
Feel free to merge after removing the commented out printf

**Pull request merged**